crategen/models.py

```diff
@@ -0,0 +1,279 @@
+from pydantic import BaseModel
+from typing import Optional
+
+
+class Executor(BaseModel):
+    """
+    A model representing an executor in the Task Execution Service (TES).
+
+    Attributes:
+        image (str): The Docker image to be used.
+        command (list[str]): The command to be executed.
+    """
+    image: str
+    command: list[str]
+
+
+class TESInputs(BaseModel):
+    """
+    A model representing input files in TES.
+
+    Attributes:
+        url (str): The URL of the input file.
+        path (str): The path where the input file should be placed.
+    """
+    url: str
+    path: str
+
+
+class TESOutputs(BaseModel):
+    """
+    A model representing output files in TES.
+
+    Attributes:
+        url (str): The URL of the output file.
+        path (str): The path where the output file is stored.
+    """
+    url: str
+    path: str
+
+
+class TESLogs(BaseModel):
+    """
+    A model representing logs in TES.
+
+    Attributes:
+        end_time (Optional[str]): The time the task ended.
+    """
+    end_time: Optional[str] = None
+
+
+class TESData(BaseModel):
+    """
+    A model representing a TES task.
+
+    Attributes:
+        id (str): The unique identifier for the TES task.
+        name (str): The name of the TES task.
+        description (Optional[str]): A brief description of the TES task.
+        executors (list[Executor]): The executors associated with the TES task.
+        inputs (list[TESInputs]): The inputs to the TES task.
+        outputs (list[TESOutputs]): The outputs of the TES task.
+        creation_time (str): The time the task was created.
+        logs (list[TESLogs]): Logs associated with the TES task.
+    """
+    id: str
+    name: str
+    description: Optional[str] = ""
+    executors: list[Executor]
+    inputs: list[TESInputs]
+    outputs: list[TESOutputs]
+    creation_time: str
+    logs: list[TESLogs]
+
+    class Config:
+        extra = "forbid"
```

Comment:
I'll update the `TESData` and `WESData` models to allow conversion from complete TES and WES objects without throwing errors. I'll also reconsider the use of `extra = "forbid" `and how it might impact this process.